The default behaviour is a fat binary, but we can specify our selection with EasyBuild

diff -rupN lightgbm-4.5.0/CMakeLists.txt lightgbm-4.5.0.new/CMakeLists.txt
--- lightgbm-4.5.0/CMakeLists.txt	2025-09-26 21:41:41.273994044 +0200
+++ lightgbm-4.5.0.new/CMakeLists.txt	2025-09-26 21:43:28.274017376 +0200
@@ -239,6 +239,8 @@ if(USE_CUDA)
         list(APPEND CUDA_ARCHS "100")
         list(APPEND CUDA_ARCHS "120")
     endif()
+    # Override the autogenerated list with one we can easily customise with EasyBuild via sed
+    set(CUDA_ARCHS EASYBUILD_CUDA_CCS)
     # Generate PTX for the most recent architecture for forwards compatibility
     list(POP_BACK CUDA_ARCHS CUDA_LAST_SUPPORTED_ARCH)
     list(TRANSFORM CUDA_ARCHS APPEND "-real")
